common/djangoapps/util/request.py

```diff
@@ -7,8 +7,8 @@
 from opaque_keys import InvalidKeyError
 from opaque_keys.edx.locations import SlashSeparatedCourseKey
 
-
-COURSE_REGEX = re.compile(r'^.*?/courses/{}'.format(settings.COURSE_ID_PATTERN))
+# accommodates course api urls, excluding any course api routes that do not fall under v*/courses, such as v1/blocks.
+COURSE_REGEX = re.compile(r'^(.*?/courses/)+(?!v[0-9]+/[^/]+){}'.format(settings.COURSE_ID_PATTERN))
```

Comment:
What is the + at the end of `(.*?/courses/)+` intended to do?  It looks like you're expecting `.*?/courses` to show up multiple times, but that seems weird.  Maybe the comment should include some examples of URLs that we intend to match with this.